API_DOCUMENTATION.md

```diff
@@ -0,0 +1,655 @@
+}
+```
+
+#### 2. Verify OTP for Purchase
+**POST** `/auth/verify-purchase-otp`
+
+Verify OTP before allowing purchase process.
+
+**Headers:** `Authorization: Bearer <JWT_TOKEN>`
+
+**Request Body:**
+```json
+{
+  "userEmail": "john@example.com",
+  "otp": "123456"
+}
+```
+
+**Response (200 OK):**
+```json
+{
+  "success": true,
+  "message": "OTP verified successfully",
+  "data": {
+    "purchaseToken": "purchase_verification_token"
+  }
+}
+```
+
+**Response (400 Bad Request):**
+```json
+{
+  "success": false,
+  "message": "Invalid or expired OTP"
+}
+```
+
+### 📦 Product Management Endpoints
+
+#### 1. Get All Products
+**GET** `/product/getAllProducts`
+
+Retrieve all products from the platform.
+
+**Response (200 OK):**
+```json
+{
+  "success": true,
+  "message": "Products fetched successfully",
+  "data": {
+    "products": [
+      {
+        "_id": "product_id",
+        "prodName": "Digital Art Pack",
+        "ProdDiscription": "High-quality digital art collection",
+        "ProdQuantity": 1,
+        "ProdImage": ["image_url"],
+        "price": 29.99,
+        "sellerId": {
+          "_id": "seller_id",
+          "UserName": "Art Seller",
+          "UserEmail": "seller@example.com"
+        },
+        "Prodcategory": "digital art",
+        "createdAt": "2024-01-15T10:00:00.000Z",
+        "updatedAt": "2024-01-15T10:00:00.000Z"
+      }
+    ],
+    "count": 1
+  }
+}
+```
+
+#### 2. Get Products by Seller
+**GET** `/product/seller/:sellerId`
+
+Retrieve all products from a specific seller.
+
+**Parameters:**
+- `sellerId`: MongoDB ObjectId of the seller
+
+**Response (200 OK):**
+```json
+{
+  "success": true,
+  "message": "Seller products fetched successfully",
+  "data": {
+    "products": [...],
+    "count": 5
+  }
+}
+```
+
+#### 3. Add Product (Sellers Only)
+**POST** `/product/addProduct`
+
+Add a new product to the platform.
+
+**Headers:** 
+- `Content-Type: multipart/form-data`
+- File upload required
+
+**Request Body (form-data):**
+```
+prodName: "My Digital Product"
+ProdDiscription: "Detailed product description"
+ProdQuantity: 1
+price: 29.99
+sellerId: "seller_mongodb_id"
+Prodcategory: "software" // Options: "software", "ebook", "music", "online courses", "digital art"
+file: [uploaded image file]
+```
+
+**Response (201 Created):**
+```json
+{
+  "success": true,
+  "message": "Product created successfully",
+  "data": {
+    "product": {
+      "_id": "product_id",
+      "prodName": "My Digital Product",
+      "ProdDiscription": "Detailed product description",
+      "ProdQuantity": 1,
+      "ProdImage": ["image_url"],
+      "price": 29.99,
+      "sellerId": "seller_id",
+      "Prodcategory": "software",
+      "createdAt": "2024-01-15T10:00:00.000Z",
+      "updatedAt": "2024-01-15T10:00:00.000Z"
+    },
+    "imageInfo": {
+      "fileName": "uploaded_file_name",
+      "directLink": "storage_url",
+      "storage": "google_drive" // or "local"
+    }
+  }
+}
+```
```

Comment:
Since we have already decided that we will be using S3 for storage purposes, why it doesnt reflect the same